test/framework/src/main/java/org/opensearch/test/engine/MockEngineFactory.java

```diff
@@ -46,6 +47,8 @@ public MockEngineFactory(Class<? extends FilterDirectoryReader> wrapper) {
 
     @Override
     public Engine newReadWriteEngine(EngineConfig config) {
-        return new MockInternalEngine(config, wrapper);
+
+        // When Segment Replication is enabled on an index replica shards use NRTReplicationEngine instead of InternalEngine.
```

Comment:
This comment explains "what" is happening (which is apparent from reading the code), but not so much "why". I'd recommend a comment more like:

> Segment replication replicas (i.e. read only replicas) do not use an InternalEngine so a MockInternalEngine will not work and an NRTReplicationEngine must be used instead. The primary shards for these indexes will still use a MockInternalEngine.

Reply:
Makes sense. Will update it. Thank you!